addons/point_of_sale/models/pos_session.py

```diff
@@ -0,0 +1,236 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class PosSession(models.Model):
+    _name = 'pos.session'
+    _order = 'id desc'
+
+    POS_SESSION_STATE = [
+        ('opening_control', 'Opening Control'),  # Signal open
+        ('opened', 'In Progress'),               # Signal closing
+        ('closing_control', 'Closing Control'),  # Signal close
+        ('closed', 'Closed & Posted'),
+    ]
+
+    def _confirm_orders(self):
+        PosOrder = self.env['pos.order']
+        for session in self:
+            company_id = session.config_id.journal_id.company_id.id
+            local_context = dict(self.env.context, force_company=company_id)
```

Comment:
You don't need to create a new context based on existing one. See [with_context definition](https://github.com/odoo/odoo/blob/0c37a8d5238a62f4bca5737398ba53d10e4f1f29/openerp/models.py#L5429)
```
# current context is {'key1': True}
r2 = records.with_context({}, key2=True)
# -> r2._context is {'key2': True}
r2 = records.with_context(key2=True)
# -> r2._context is {'key1': True, 'key2': True}
```
so in you case `PosOrder.with_context(force_company=company_id)._create_account_move...` is enough